### DISCUSSION TITLE
GrapesJS integration with Alpine JS / Custom HTML tags

### DISCUSSION THREAD

**Discussion by daniel23ad-pulse** [GrapesJS/grapesjs#5272] [Q&A]
While working on a certain project the idea ocurred of using both GrapesJS and Alpine JS. 
So, using GrapesJS's Block Manager I added a component type and its HTML to GrapesJS's editor and it works if I use any other HTML that does not use the template HTML tag.

![Screenshot 2023-08-03 at 10 24 59](https://github.com/GrapesJS/grapesjs/assets/141316746/c0bc6d2e-5526-477c-bcf6-fe9ad4790aba)

So here, both the first and second components work, as they have fairly simple HTML associated with them. In the third component, as I wanted to pass something more complex and dynamic by using AlpineJS's functionalities, it does not work so well. Apart from that specific HTML template tag, everything seems to be working fine. The only thing is that when I try to render a tag like this in GrapesJS, the only thing i get inside of it is: '#document-fragment'... here is an example:

What I am getting:
![Screenshot 2023-08-03 at 10 30 43](https://github.com/GrapesJS/grapesjs/assets/141316746/eed97499-8249-4e02-b369-a055faa4d35e)

What I am passing to GrapesJS (which should render 2 spans inside the template tag):
![Screenshot 2023-08-03 at 10 32 45](https://github.com/GrapesJS/grapesjs/assets/141316746/6c513550-fdd3-4fd7-9fa7-dacc8ccada3b)

I already tried [this](https://github.com/GrapesJS/grapesjs/issues/565) adapted to the tag I am having problems with but that way I get an AlpineJS error as it does not allow the development of 'x-for' cycles outside of template tags. Is this something that's possible to solve within GrapesJS or should I complain also to the Alpine JS developers? 
 
Thanks in advance!

**Comment by c-nv-s** [comment]
interested in a solution for integrating alpinejs as well

**Comment by artf** [comment]
There is a [change](https://github.com/GrapesJS/grapesjs/commit/a7d4942e57ebf43344fe386b49a425f585fb25e5) which should add the proper support to `<template>` tags. Will be available in the next release.

**Comment by bigdududu** [comment]
met the same issue, have you solved this?
